abi/src/test/java/org/web3j/abi/AbiV2TestFixture.java

```diff
@@ -117,6 +118,12 @@ public class AbiV2TestFixture {
     public static final String FUNC_setArrayOfStructWithArraysFunction =
             "setArrayOfStructWithArraysFunction";
 
+    // TODO Shall we reconsider const naming convention and unified constants here?
```

Comment:
Please note ```//TODO``` I left here. What would be your opinion on this?

Reply:
Certainly you could, and that way is more readable, if you open a PR, I would :+1: for it, but would need one of the maintainers to approve it and approve the renaming.

Thanks a lot for the time you're giving the library.

Reply:
You are welcome. Thank you for all your work, otherwise I have to start all work on the similar library since the beginning. :)

Reply:
Generally, if you already have your smart contract written and your structs defined, you could either fork the code and tweak it to work for your case, or you could even write an encoder/decoder for the specific structs you have from scratch easily.

Reply:
Forking a codebase would lead to constant sync with origin repo. In case of multiple commits it means possible conflict resolution and parallel library maintaining. I thought about this, but left it as a last resort.

Thank you for point out on flexibility in customisation of type encoder\decoder. I will look at this. If you have this documented somewhere, it would be good to see links too.

Reply:
There is unfortunately no documentation for this, I haven't checked in a while tho.

But generally, I based my work on [this](https://docs.soliditylang.org/en/v0.8.17/abi-spec.html). If you manage to understand how the encoding/decoding works for static types and dynamic types, you could easily write an encoder/decoder specific to certain structs. That is because you already know what to expect and the fields offsets.

To test your knowledge on the offsets, check the tests scenarios [here](https://github.com/web3j/web3j/blob/master/abi/src/test/java/org/web3j/abi/FunctionReturnDecoderTest.java). They provide encodings + the corresponding structs. That should be a good starting point.